### PR TITLE
Add static ALFF and ReHo plots to report

### LINKED ISSUE #626: [ENH] Add mosaic plots for ALFF and ReHo

### PR DESCRIPTION
**Pull request by kahinimehta**
Closes #626.

Add mosaic plots and integrate them into the report

### CODE REVIEW

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1067]: I don't think you need ReHo- and ALFF-specific functions for this. A generic `plot_stat_map` call should be fine.

**Comment by tsalo** [comment]
You should be able to use mosaic plots once we merge #607, as that PR bumps the pinned nilearn version to 0.9.2.

File: xcp_d/interfaces/resting_state.py
Comment (tsalo) [inline xcp_d/interfaces/resting_state.py:10]: `shutil` is a builtin module, so it should go up with `os`, not with the third-party modules like `nilearn` and `nipype`.

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py]:
```suggestion
    resolution = parse_file_entities(bold_file).get("res")
```

Since the parsed file entities should be a dictionary, you can just get `.get`, which will default to None if the key isn't in the dictionary.

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1017]: Why convert this to a string? What type is it by default?

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1069]:
```suggestion
    density = parse_file_entities(bold_file).get("den", "32k")
```

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1138]: You don't need to duplicate a variable like this.

```suggestion
    return output_path
```

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1021]:
```suggestion
    resolution = parse_file_entities(bold_file).get("res", "1")
```

The output from `parse_file_entities` should be a dictionary, so you can just use `.get()`, which will return None by default if the key isn't present in the dictionary, or you can set the value it will return via the second parameter (e.g., `.get(key, default_value)`).

**Review by tsalo — CHANGES_REQUESTED** [review]
Some changes.

File: xcp_d/workflow/restingstate.py
Comment (tsalo) [inline xcp_d/workflow/restingstate.py:126]: Can be simplified, since the only difference is the function.

```suggestion
    alff_plot = pe.Node(Function(
                        input_names=["output_path", "filename", "bold_file"],
                        output_names=["output_path"],
                        function= plot_alff_reho_surface  if cifti else plot_alff_reho_volumetric),
                        name="alff_plot")
    alff_plot.inputs.output_path = 'alff.svg'
    alff_plot.inputs.bold_file = bold_file
```

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1063]: You don't need to do this.

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1071]: I'd grab the space from the file_entities like you do with the volumetric function, since we plan to support fsaverage in the future.

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1131]: Probably need to get the absolute path before returning `output_path`.

**Review by tsalo — CHANGES_REQUESTED** [review]
Can you drop the changes to `data/fmriprep/sub-colornest001/.../` and remove the added crash file? Can you also add the new plots to the report config file?

File: xcp_d/interfaces/bids.py
Comment (tsalo) [inline xcp_d/interfaces/bids.py:135]: Why move this import down from the top of the file?

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1013]: Can you fill out the numpydoc sections as well (Parameters, Returns, optionally Notes).

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1067]: Should have an empty line between the imports and any code.

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1032]:
```suggestion
                        display_mode='mosaic',
```

Since we bumped the pinned nilearn version, you can use mosaic here.

File: xcp_d/data/reports.yml
Comment (tsalo) [inline xcp_d/data/reports.yml:34]:
```suggestion
    subtitle: Framewise Displacement and Censored Volumes
```

File: xcp_d/interfaces/bids.py
Comment (kahinimehta) [reply xcp_d/interfaces/bids.py:135]: It wasn't being imported unless I moved it into the function

File: xcp_d/utils/plot.py
Comment (kahinimehta) [reply xcp_d/utils/plot.py:1032]: sounds good!

File: xcp_d/data/fmriprep/sub-colornest001/figures/sub-colornest001_ses-1_task-rest_run-1_desc-validation_bold.html
Comment (tsalo) [inline xcp_d/data/fmriprep/sub-colornest001/figures/sub-colornest001_ses-1_task-rest_run-1_desc-validation_bold.html:1]: There's still a diff here.

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1125]: Optimally, it would be nice to have a colorer, but I'm not sure how to do that without making everything look terrible. Maybe it can be an open issue. In the meantime, we should at least ensure that the minimum and maximum values are the same across the four subplots. Try extracting the minimum and maximum values across both hemispheres' surface data and then using those to set vmin and vmax in all of the subplots.

File: xcp_d/utils/plot.py
Comment (tsalo) [inline xcp_d/utils/plot.py:1116]:
```suggestion
    v_max = np.max([np.max(lh_surf_data), np.max(rh_surf_data)])
    v_min = np.min([np.min(lh_surf_data), np.min(rh_surf_data)])
```

Just so it only takes the surface data into account. The lines would need to go after the surface data are loaded though.

File: xcp_d/interfaces/resting_state.py
Comment (tsalo) [inline xcp_d/interfaces/resting_state.py:106]: Seems like a misnomer. The variable and description indicate it's a directory, but the class is a File. I still think you should just hardcode the filename within the plotting function though.

File: xcp_d/workflow/cifti.py
Comment (tsalo) [inline xcp_d/workflow/cifti.py:565]: Maybe just rehoplot. No need to be too specific.

File: xcp_d/interfaces/resting_state.py
Comment (kahinimehta) [reply xcp_d/interfaces/resting_state.py:106]: not sure why that was even there

**Review by tsalo — CHANGES_REQUESTED** [review]
I just have some requests for the captions and that modified HTML file.

File: xcp_d/data/fmriprep/sub-colornest001/figures/sub-colornest001_ses-1_task-rest_run-1_desc-validation_bold.html
Comment (kahinimehta) [reply xcp_d/data/fmriprep/sub-colornest001/figures/sub-colornest001_ses-1_task-rest_run-1_desc-validation_bold.html:7]: Hmm, not sure why this keeps happening - will try replacing the file with the original version

File: xcp_d/data/reports.yml
Comment (kahinimehta) [reply xcp_d/data/reports.yml]: added some info about the overlays! what else might be good to include here?

Comment (tsalo) [reply xcp_d/data/reports.yml]: Thanks. That looks good to me.

File: xcp_d/workflow/cifti.py
Comment (tsalo) [inline xcp_d/workflow/cifti.py:566]: This needs to depend on `bandpass_filter`

File: xcp_d/workflow/cifti.py
Comment (tsalo) [inline xcp_d/workflow/cifti.py:581]: This needs to depend on `bandpass_filter`

**Review by tsalo — APPROVED** [review]
Thanks! The code LGTM, so once CI passes and you've taken a quick look at the generated reports to make sure they don't look wonky, I think you can merge this.

**Pull request merged**
